frontend/src/components/authentication/ForgotPassword.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useState, useCallback} from 'react'
+import {CssBaseline, TextField} from '@material-ui/core'
+import {Button, Avatar, Typography, Link, Paper, Grid, Box }  from '@material-ui/core'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
+
+import * as gridSize from '../../constants/styles.values'
+import {SIX as SHADOW_DEPTH_SIX} from '../../constants/styles.values'
+import { emailFormatIsValid } from '../../utils/FormatVerifications'
+import {useStyles} from './SignUp.styles'
+import { AuthApiClient } from '../../api/authRequest'
+
+export default function ForgotPasswordSide(): React.ReactElement {
+
+  const classes = useStyles()
+
+  const initialState = {
+    email: '',
+    message: '',
+    error: false
+  }
+
+  const [state, setState] = useState(initialState)
+  
+  const onChangeEmail = useCallback((e) => {
+    setState({...state, email: e.target.value})
+  }, [setState, state])
+  
+  const recoverPassword = useCallback((e) => {
+    e.preventDefault()
+    if (emailFormatIsValid(state.email)) {
+      const authApiClient = new AuthApiClient()
+      authApiClient.recover(state.email).then((res) => {
```

Comment:
Curious - why not asyn/await?

Reply:
Because it works :) And really I think it gives the same result.
An await makes a function synchronous again. Meaning that the code after the await statement will be executed after this statement.
In this code that I wrote, instead of await I do a callback, which also executes some code after the async function.
Correct me if I'm wrong.

I changed the function to async/await if you like it better.